infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -286,10 +286,12 @@ private void prefixTrim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext
     @ServerHandler(type = CorfuMsgType.READ_REQUEST)
     private void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
         long address = msg.getPayload().getAddress();
-        log.trace("read: {}", msg.getPayload().getAddress());
+        boolean cacheable = msg.getPayload().isCacheable();
+        log.trace("read: {}, cacheable: {}", msg.getPayload().getAddress(), cacheable);
+
         ReadResponse rr = new ReadResponse();
         try {
-            ILogData e = dataCache.get(address);
+            ILogData e = cacheable ? dataCache.get(address) : handleRetrieval(address);
```

Comment:
Suppose the value to be read is already in the cache, then this will actually result in a slowdown because handleRetrieval will go to the fileChannel anyway right? Instead instead of directly going to handleRetrieval, can we do that only if dataCache.getIfPresent() returns null?

Reply:
Done.